lib/jxl/jxl_test.cc

```diff
@@ -405,6 +405,43 @@ TEST(JxlTest, RoundtripMultiGroup) {
               IsSlightlyBelow(18));
 }
 
+TEST(JxlTest, RoundtripRGBToGrayscale) {
+  ThreadPoolInternal pool(4);
+  const PaddedBytes orig = ReadTestData("jxl/flower/flower.png");
+  CodecInOut io;
+  ASSERT_TRUE(SetFromBytes(Span<const uint8_t>(orig), &io, &pool));
+  io.ShrinkTo(600, 1024);
+
+  CompressParams cparams;
+  cparams.butteraugli_distance = 1.0f;
+  cparams.speed_tier = SpeedTier::kFalcon;
+
+  extras::JXLDecompressParams dparams;
+  dparams.color_space = "Gra_D65_Rel_SRG";
+
+  CodecInOut io2;
+  EXPECT_FALSE(io.Main().IsGray());
+  EXPECT_LE(Roundtrip(&io, cparams, dparams, &pool, &io2), 55000u);
+  EXPECT_TRUE(io2.Main().IsGray());
+
+  // Do a very crude conversion to grayscale and compare with butteraugli.
+  Image3F* color = io.Main().color();
+  for (size_t y = 0; y < color->ysize(); ++y) {
+    float* row_r = color->PlaneRow(0, y);
+    float* row_g = color->PlaneRow(0, y);
```

Comment:
use planes 0, 1, 2?

Reply:
Thanks for catching this! The test passed because both the test and the implementation just dropped green and blue ... Now it works and butteraugli distance also improved.